src/helpers/helpers.canvas.js

```diff
@@ -5,7 +5,7 @@ import {calculateTextAlignment, getSize} from './helpers.options';
 const widthCache = new Map();
 
 export function isImageOrCanvas(content) {
-  return content instanceof Image || content instanceof HTMLCanvasElement;
+  return content && (content.toString() === '[object HTMLCanvasElement]' || content.toString() === '[object HTMLImageElement]');
```

Comment:
If the content is a string containing '[object HTMLCanvasElement]' or '[object HTMLImageElement]', this will think its image or canvas. I'm also not sure if everything non-false contains a `.toString()` method.

Another solution is to detect the DOM availability separately and change the function signature based on that:

```js
const isDOMSupported = typeof window !== 'undefined' && typeof document !== 'undefined';

export const isImageOrCanvas = isDOMSupported 
  ? (content) => content instanceof Image || content instanceof HTMLCanvasElement 
  : (content) => content instanceof Image;
```
I'm also not sure if `Image` is available in node always.

I think there are quite a few other possible solutions too.

Reply:
I had this doubt but I see that in `Chart.js`, [helpers.canvas](https://github.com/chartjs/Chart.js/blob/1c837a9c12ec4fa7c519a170830a4adbb9898d09/src/helpers/helpers.canvas.js#L125-L142), we are doing the checking on the `toString` method:

```javascript
export function drawPoint(ctx, options, x, y) {
  let type, xOffset, yOffset, size, cornerRadius;
  const style = options.pointStyle;
  const rotation = options.rotation;
  const radius = options.radius;
  let rad = (rotation || 0) * RAD_PER_DEG;


  if (style && typeof style === 'object') {
    type = style.toString();
    if (type === '[object HTMLImageElement]' || type === '[object HTMLCanvasElement]') {
      ctx.save();
      ctx.translate(x, y);
      ctx.rotate(rad);
      ctx.drawImage(style, -style.width / 2, -style.height / 2, style.width, style.height);
      ctx.restore();
      return;
    }
  }
```

Reply:
that means that also Chart.js couldn't work correctly.

Reply:
No, because i has an additional typeof check, that makes sure its an object before comparing to string.

Reply:
This should work (though its calling toString twice if its a Image):
```js
return typeof content === 'object' && (content.toString() === '[object HTMLCanvasElement]' || content.toString() === '[object HTMLImageElement]');
```

Reply:
> I'm also not sure if `Image` is available in node always.

it seems available: https://github.com/Automattic/node-canvas/blob/master/src/Image.cc

Reply:
Applied your suggested updates